client/ayon_nuke/api/plugin.py

```diff
@@ -158,7 +162,7 @@ def create_instance_node(
 
         return created_node
 
-    def set_selected_nodes(self, pre_create_data):
+    def _set_selected_nodes(self, pre_create_data):
```

Comment:
I have a question, why we do store selected nodes to the object of create plugin? Wouldn't be better to have helper methods that would just return then?

```

Reply:
So I gave a try to this approach as indeed it make things simpler, but actually the necessity to store selected file raised immediatly. A lot of the creators manipulate the node graph during their instance(s) creation and this lost the intial selected nodes.
I'm adding a comment to reflect this.

Reply:
Well, shouldn't that be called once in create plugin (at the start), and then pass the node where it is needed. So we don't rely on `self.selected_node`, which might be changed in some method -> it is really hard to find where it is really set and keep track of what is stored there at the moment it is used.

Reply:
I reworked the node selection logic so we no more rely on the `self.selected_node` class properties. 
Please let me know if that corresponds better to what you had in mind.